### ISSUE TITLE
Generating Videos with Dynamics-aware Implicit Generative Adversarial Networks

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#150]
## TL; DR

- Employ Implicit Neural Representation (INR-GAN, to be specific) for video generation task
- Use image level convolutional discriminator with extra time-conditioning for efficient motion discrimination.

## Motivation

- Video generation suffers from heavy computational burden due to cubic memory / computation requirements.
- Traditional 3D-grid based convolutional generative models fail to fully leverage temporal correlation of videos.

### Video Generators

- MoCoGAN
- DVD-GAN

### INR-based Generative Model

- INR-GAN (Adversarial Generation of Continuous Images)

## Method

![image](https://user-images.githubusercontent.com/46308407/152273522-4a600071-866a-4757-b73b-f5df76b78fde.png)

- Generally build upon INR-GAN (CVPR'21)
- Generator samples weights for the video-INR (hypernetwork)
- Two generators: content generator(x,y coordinate) + motion generator(t coordinate)

### Generator

- Add motion latent vector
- Use lower frequency sinusoidal encoding for time coordinates (inductive bias: temporal shift < spatial shift)
- Additional non-linear mapping on top of generated motion(temporal) weights

### Discriminator

- Same discriminator as INR-GAN
- Need additional motion discriminator commonly used in video-GANs (3D conv networks if usual)
- Since DIGAN(Dynamics-aware Implicit GAN) can generate frames at arbitrary time step (input: x, y, t), motion discriminator functions in similar manner
- Takes in (Image1, Image2, t2-t1) as input and looks for unnatural motion artifacts

### Advantages

- Lower computational burden (fixed network size for INR)
- Natural employment of temporal correlation
- Temporal interpolation/extrapolation
- Spatial interpolation/extrapolation (zoom-in/out)

![image](https://user-images.githubusercontent.com/46308407/152274035-cd780920-6072-4b0f-b58b-3123fb9fae05.png)
![image](https://user-images.githubusercontent.com/46308407/152274049-bf57e19e-112a-4993-a9bc-37a669c141a7.png)
![image](https://user-images.githubusercontent.com/46308407/152274056-a4e95546-1a13-4d19-80bb-3b25458e8fe7.png)
![image](https://user-images.githubusercontent.com/46308407/152274072-66e5803b-03ee-44a1-88da-7a16ce2297fb.png)
![image](https://user-images.githubusercontent.com/46308407/152274086-30178d68-1055-4b05-b92f-750f3220127c.png)

## Comments
- Clear motivation and simple (yet effective) methods proposed.
- Timely incorporate INR framework to video generation task.
- Use of INR in video generation task does make a lot of sense, more than in image generation tasks.
- Although a significant progress, generated samples indicate there's still a lot more to go for the challenging task of video synthesis.
- Personally makes me a little jealous how the authors leveraged a set of such simple ideas and composed it into a decent paper.
